heron/tmaster/src/cpp/manager/stateful-checkpointer.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __TMASTER_STATEFUL_CHECKPOINTER_H_
+#define __TMASTER_STATEFUL_CHECKPOINTER_H_
+
+#include <string>
+#include <unordered_set>
+#include "network/network.h"
+#include "proto/tmaster.pb.h"
+#include "manager/tmaster.h"
+#include "basics/basics.h"
+
+
+namespace heron {
+namespace proto {
+class PhysicalPlan;
+}
+}
+
+namespace heron {
+namespace tmaster {
+
+class StatefulCheckpointer {
```

Comment:
please add class-level docs describing what this class does and what it's responsibilities are.